File: clients/php/src/Result/StoreResult.php
<?php

declare(strict_types=1);

namespace Priostack\Acn\Result;

/**
 * Result of {@see \Priostack\Acn\Client::store()}.
 */
final class StoreResult
{
    /**
     * @param list<array<string, mixed>> $objectRefs One ref per ingested object.
     * @param array<string, mixed>       $raw
     */
    public function __construct(
        public readonly array $objectRefs,
        public readonly array $raw = [],
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromData(array $data): self
    {
        return new self(
            objectRefs: array_values((array) ($data['objectRefs'] ?? [])),
            raw: $data,
        );
    }

    /** How many objects were ingested. */
    public function count(): int
    {
        return count($this->objectRefs);
    }
}

File: clients/java/src/main/java/com/priostack/acn/StoreResult.java
package com.priostack.acn;

import com.fasterxml.jackson.databind.JsonNode;
import java.util.List;

/** Result of {@link ACNClient#store}. */
public final class StoreResult {

    private final List<JsonNode> objectRefs;
    private final JsonNode raw;

    public StoreResult(List<JsonNode> objectRefs, JsonNode raw) {
        this.objectRefs = objectRefs;
        this.raw = raw;
    }

    /** One reference per ingested object. */
    public List<JsonNode> objectRefs() {
        return objectRefs;
    }

    /** How many objects were ingested. */
    public int count() {
        return objectRefs.size();
    }

    public JsonNode raw() {
        return raw;
    }

    @Override
    public String toString() {
        return "StoreResult{count=" + count() + '}';
    }
}
